libs/blocks/merch/merch.js

```diff
@@ -1031,6 +1031,38 @@ export async function getCommerceContext(el, params) {
   return { promotionCode, perpetual, wcsOsi };
 }
 
+// TODO: remove this function once fallbackStep is fully authored
+function getFallbackStep(wcsOsi, modal, checkoutClientId) {
+  if (checkoutClientId !== 'doc_cloud') return undefined;
+  const MODAL_TYPE_3_IN_1 = {
+    TWP: 'twp',
+    D2P: 'd2p',
+    CRM: 'crm',
+  };
+  const is3in1Modal = Object.values(MODAL_TYPE_3_IN_1).includes(modal);
+  if (!is3in1Modal) return undefined;
+  const masFF3in1 = document.querySelector('meta[name=mas-ff-3in1]');
+  const is3in1Enabled = !masFF3in1 || masFF3in1.content !== 'off';
+  if (is3in1Enabled) return undefined;
+  const osiToStepMap = {
+    'vQmS1H18A6_kPd0tYBgKnp-TQIF0GbT6p8SH8rWcLMs': 'commitment',
```

Comment:
Is the list fixed? Can we get it from a JSON instead?

Reply:
This is temporarily hardcoded so that authors have time to author proper workflow steps for each CTA. I would not place it in JSON, as this function will be removed right after authors finish with their changes